crates/evm/evm/src/executors/mod.rs

```diff
@@ -48,8 +48,17 @@ pub use trace::TracingExecutor;
 
 sol! {
     interface ITest {
+        #[derive(Default)]
+        struct BeforeTestSelectors {
+            bytes4 test_selector;
+            bytes4[] before_selectors;
+        }
+
         function setUp() external;
         function failed() external view returns (bool failed);
+
+        #[derive(Default)]
+        function beforeTestSelectors() public view returns (BeforeTestSelectors[] memory beforeSelectors);
```

Comment:
What about `(bytes4) -> bytes[]`?
1. selector as input to function, can be ignored or grouped with others programmatically
2. bytes[] as return instead of bytes4[] for arbitrary calldata

Reply:
that would be nice, I followed the way `FuzzSelector (addr -> bytes4[])` and `function targetSelectors() public view returns (FuzzSelector[]` is implemented to have kind of consistency but can switch, should we care about having them similar?

Reply:
actually that's awesome so no concern re any consistency  :) pls check https://github.com/foundry-rs/foundry/pull/8497/commits/bf8960358e89bf89a77289d76483698e20280cc7